File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var map = new Map()
    
    for(const name of participant){
        map.set(name, map.get(name) + 1 || 1)
    }
    
    for(const name of completion){
        map.set(name, map.get(name) - 1)
    }

    for(const [a, b] of map){
        if(b > 0){
            return a
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    arr = {}
    
    for p in participant :
        arr[p] = arr.get(p, 0) + 1
    for c in completion :
        arr[c] -= 1
    
    for name, count in arr.items() :
        if count > 0 :
            return name
        
    
    return answer